securetar/__init__.py

```diff
@@ -587,6 +587,16 @@ def _atomic_contents_add(
             _atomic_contents_add(tar_file, directory_item, file_filter, item_arcname)
             continue
 
-        tar_file.add(directory_item.as_posix(), arcname=item_arcname, recursive=False)
+        _add_wrap_error(tar_file, directory_item, item_arcname)
 
     return None
+
+
+def _add_wrap_error(tar_file: tarfile.TarFile, file_path: Path, arcname: str) -> None:
+    """Wrap add function with error handling."""
+    try:
+        tar_file.add(file_path.as_posix(), arcname=arcname, recursive=False)
+    except (OSError, tarfile.TarError) as err:
+        raise SecureTarError(
```

Comment:
Should we maybe have a custom exception with file path as field? :thinking:

Reply:
Yes, that's a good idea 👍 
I was also thinking we might as well catch and wrap `Exception` instead of selecting only `(OSError, tarfile.TarError)`?